src/resource/hal-resource-client.factory.js

```diff
@@ -201,11 +228,51 @@ export default function HalResourceClientFactory($q, $injector, $halConfiguratio
     function toStringItem(item) {
       return item.toString();
     }
+
+    /**
+     * Execute a HTTP GET request on self
+     *
+     * @param {String}      rel
+     * @param {Object|null} urlParams
+     * @param {Object}      options
+     * @return {Promise}
+     */
+    function $reloadSelf(urlParams, options) {
```

Comment:
Reload self is not really reloading anything. It's making a request and returning a new resource. Can we call it `$getSelf` to comply with the other self methods?